test/functional/test_framework/test_node.py

```diff
@@ -377,17 +377,23 @@ def is_node_stopped(self):
             return False
 
         # process has stopped. Assert that it didn't return an error code.
-        assert return_code == 0, self._node_msg(
-            "Node returned non-zero exit code (%d) when stopping" % return_code)
+        # unless 'expected_ret_code' is provided.
+        if expected_ret_code is not None:
+            assert return_code == expected_ret_code, self._node_msg(
+                "Node returned unexpected exit code (%d) vs (%d) when stopping" % (return_code, expected_ret_code))
```

Comment:
nit, could take the opportunity to change to f-strings here (and below), as [suggested in the functional test style guidelines](https://github.com/bitcoin/bitcoin/blob/361a0c00b3fdac7149c027c2c69e4db60ee5aa86/test/functional/README.md?plain=1#L39):
```suggestion
                 f"Node returned unexpected exit code ({return_code}) vs ({expected_ret_code}) when stopping")
```